### ISSUE TITLE
Add Multiple Pages Does Not Trim Leading Space(s)

### ISSUE THREAD

**Issue by jeremy-farrance** [dnnsoftware/Dnn.Platform#3966]
Note: I understand if this gets changed to a Feature Request, but I do consider it a bug and it is also a very time consuming mistake to have to go fix after it occurs on a few dozen new pages at once.

## Description of bug
When using the PersonaBar Pages, Add Multiple Pages feature, putting a space at the start of a page name or in between the indents (greater-then symbol) causes unexpected results and odd URLs.

## Steps to reproduce
1. Go to PersonaBar, Content, Pages
2. Click on (upper right) the Add Multiple Pages button
3. Enter the following in to BULK PAGES TO ADD; its important to include the extra spaces
```
Top Page
> Child Page 1
> Child Page 2
> > Child SubPage 1
>> Child SubPage 2
```
4. Click the Add Page(s) button
5. Expand the new Top Page to see the results

**Problem 1** - Child SubPage 1 is under Top Page, not where it belongs
 
6. Expand Child Page 2, click on Child SubPage 2 and `...` View that page

**Problem 2** - Look at the URL? The page names retained the leading space between ">" and "Child..." and caused DNN to then replace the space with a dash in the generated URL:

```
/Top-Page/-Child-Page-2/-Child-SubPage-2
----------^  and -------^
```

## Current behavior
For Problem 2, the code that processes the BULK text entered needs to strip/trim the spaces when passing the intended page name along to each Add Page call.
 
For Problem 1, when processing multiple indents, spaces should be stripped first so that `">>>"` or `" >   > > "` or whatever properly parses 3 levels instead of the current behavior which for the latter, would only process 1 level and add `"  > > "` to the front of the page name.

## Expected behavior
These are usually typed by hand and it is more common to type `"> Page Name"` (with the space) instead of `">Page Name"` and in either case, the new page should not have a leading space in the name, e.g. `" Page Name"` and then end up with a URL of `"...site.com/-Page-Name"`.

## Screenshots
SiteMap output showing /-Fred
![image](https://user-images.githubusercontent.com/653301/89692246-c057d400-d8d0-11ea-88ea-34f157ebdc75.png)

## Context
A client did this while creating 40+ pages in 3 levels of hierarchy. It wasn't brought to our attention as a problem until 2 weeks later, so restore from backup was not an option. Granted, an intern fixed it. But it took a couple of very tedious hours to go page by page fixing the name and the SEO/URL for each page.
 
## Affected version
* [x] 09.06.02 
* [x] all versions since PersonaBar was added

## Affected browser
any/all

**Comment by mitchelsellers** [comment]
Just my $0.02 here, the current behavior is expected, and trying to "fix" this could get really sticky.

Considering your request `> > Child SubPage 1` This is not a sub-sub page, this is a sub-page with a requested page name of `> Child SubPage 1`. This would be no different than say `> ! Child SubPage 1` or similar.

The system doesn't have any way to know that > > was really supposed to be >>, so I'm not sure how we address this.

Now, for Problem 2, I'm a bit indifferent on the URL here, this is a bit of a "replacement token" issue, when you have a non-URL friendly character at the beginning, maybe we should change that.  But this would be a breaking change

**Comment by jeremy-farrance** [comment]
> Considering your request `> > Child SubPage 1` This is not a sub-sub page, this is a sub-page with a requested page name of `> Child SubPage 1`. This would be no different than say `> ! Child SubPage 1` or similar.

Please re-consider what you wrote - as a non-programmer - as new user just learning the system. I don't think you would find many people that would agree with what you just said. This is a simple parsing issue that just needs a little TLC to be more forgiving of wonderfully intuitive humans who learn as they go and refuse to read docs (which in this case are sparse).

**Comment by mitchelsellers** [comment]
I understand at a high level your point, but if we allow arbitrary space removal, how do we prevent removal of spaces when we shouldn’t. Or provide consistency. 

I’m fine being overridden here, but programmer or not the system just magically removing or adjusting user input on “assumption” which is what this would be never seems to work well in the end. This has been my experience. 

@david-poindexter this is a good one for you

**Comment by david-poindexter** [comment]
The handling of whitespaces in page names, as it relates to the resulting URL, is handled as a setting elsewhere.  Allowing spaces in page names is 100% allowed (even leading and trailing).  Stripping out these would not only be a breaking change, but it would no longer allow admins the flexibility they currently have.

Now, regarding the greater-than symbols, I could see an opportunity for improvement here.  Since these are not valid URL characters, we could perhaps strip any space(s) between two ">" to provide a slightly better UX.  We could even consider stripping leading space(s).  However, I would not recommend stripping any trailing space(s) because that is part of the page name.

Bottom line, this is a admin power feature and we should really just provide better inline help and documentation for proper syntax if it doesn't already exist.  

Also, the good news is, it only takes one or two times doing it "wrong" to want to learn the "right" way!  😂👍🏼

**Comment by jeremy-farrance** [comment]
Then the better option might be to enhance the "nearby" (on-page, self-) documentation and maybe consider at least catching these gotchas in the Validate option?
 
Sidenote: I tested and we already strip trailing spaces, so please don't undo that. :)

**Comment by jeremy-farrance** [comment]
> Also, the good news is, it only takes one or two times doing it "wrong" to want to learn the "right" way! 😂👍🏼

“The burned hand teaches best. After that, advice about fire goes to the heart.” - J.R.R. Tolkien

**Comment by sleupold** [comment]
I agree with @jeremy-farrance that the "add multiple pages" feature should not support whitespaces, as this would be a typical issue for non-experienced users. We should trim page names (even when adding single pages), because it would be an edge case to support it, which could still be handled using Prompt or SQL, but normally users would not expect pages to start or end with a blank or dash.

**Comment by bdukes** [comment]
Is there a valid use case for leading or trailing whitespace in a page name?  It seems like, especially in the add multiple pages scenario, that it wouldn't be what a user wanted.

**Comment by mitchelsellers** [comment]
Just want to be clear here that we are talking dual purpose here. 

Page name becomes URL. 

If you end a page with a non-url character you can get a trailing space in the URL. 

Such as. 

Add Item +

Or 

Read More >

**Comment by jeremy-farrance** [comment]
Yes, but not spaces, if you enter `">Page Name   "` you will not get trailing dashes like "/Page-Name---"

**Comment by jeremy-farrance** [comment]
I just came back and reread things. It was a mistake for me to put "Whitespace" in the title of the issue. I went ahead and changed that just now. Sorry for the confusion.
